backend/src/main/java/com/happy/friendogly/notification/service/DeviceTokenService.java

```diff
@@ -0,0 +1,38 @@
+package com.happy.friendogly.notification.service;
+
+import com.happy.friendogly.member.repository.MemberRepository;
+import com.happy.friendogly.notification.domain.DeviceToken;
+import com.happy.friendogly.notification.dto.request.UpdateDeviceTokenRequest;
+import com.happy.friendogly.notification.dto.response.UpdateDeviceTokenResponse;
+import com.happy.friendogly.notification.repository.DeviceTokenRepository;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DeviceTokenService {
+
+    private final DeviceTokenRepository deviceTokenRepository;
+
+    private final MemberRepository memberRepository;
+
+    public DeviceTokenService(DeviceTokenRepository deviceTokenRepository, MemberRepository memberRepository) {
+        this.deviceTokenRepository = deviceTokenRepository;
+        this.memberRepository = memberRepository;
+    }
+
+    public UpdateDeviceTokenResponse update(Long memberId, UpdateDeviceTokenRequest updateDeviceTokenRequest) {
+        Optional<DeviceToken> optionalDeviceToken = deviceTokenRepository.findByMemberId(memberId);
+        if (optionalDeviceToken.isPresent()) {
+            DeviceToken deviceToken = optionalDeviceToken.get();
+            deviceToken.changeDeviceToken(updateDeviceTokenRequest.deviceToken());
+            return new UpdateDeviceTokenResponse(deviceToken.getDeviceToken());
+        }
+        DeviceToken deviceToken = deviceTokenRepository.save(
+                new DeviceToken(
+                        memberRepository.getById(memberId),
+                        updateDeviceTokenRequest.deviceToken()
+                        )
+        );
+        return new UpdateDeviceTokenResponse(deviceToken.getDeviceToken());
+    }
```

Comment:
```suggestion
    public UpdateDeviceTokenResponse update(Long memberId, UpdateDeviceTokenRequest updateDeviceTokenRequest) {
        DeviceToken deviceToken = deviceTokenRepository.findByMemberId(memberId)
                .map(token -> {
                    token.changeDeviceToken(updateDeviceTokenRequest.deviceToken());
                    return token;
                })
                .orElse(deviceTokenRepository.save(
                        new DeviceToken(memberRepository.getById(memberId), updateDeviceTokenRequest.deviceToken())));

        return new UpdateDeviceTokenResponse(deviceToken.getDeviceToken());
    }
```

map, orElse 써서 약간의 개선이 가능할 것 같아요.

Reply:
(수정) orElseGet을 써야 하는군요... !!!!